lib/user_agent_parser/version.rb

```diff
@@ -45,9 +47,24 @@ def eql?(other)
         version == other.version
     end
 
-    alias == eql?
+    def ==(other)
+      other = normalize_version(other)
+
+      self.<=>(other).zero?
+    end
+
+    def <=>(other)
+      other = normalize_version(other)
+
+      range = [segments.size, other.segments.size].min - 1
+
+      (0..range).each do |i|
```

Comment:
I would use https://github.com/canton7/sem_version instead of rewriting your own comparable for version and you'll get everything for free

Reply:
FWIW `Gem::Version` is [capable of comparing versions](https://stackoverflow.com/questions/2051229/how-to-compare-versions-in-ruby). It's already available without  an additional dependency and it even works with non-SemVer-versioning.

Reply:
The problem I'm having with `Gem::Version` is that it does not consider `9` equal to `9.1.2`, i.e. I can't easily test for Internet Explorer 9 all versions.